it/flags-provider/src/test/java/com/linecorp/armeria/common/FlagsProviderTest.java

```diff
@@ -127,6 +127,14 @@ void twoRequestContextStorageProvidersAreProvidedAndInvalidFQCNisSpecify() throw
         assertThat(actual).isEqualTo(Custom1RequestContextStorageProvider.class.getSimpleName());
     }
 
+    @Test
+    @SetSystemProperty(key = "com.linecorp.armeria.tlsEngineType", value = "jdk")
```

Comment:
Note: technically tests run with parallel may be affected by this.  (we run our builds with the `--parallel` flag)
Having said this, it seems like we already us `SetSystemProperty`, so we may handle this in bulk another time.

ref: https://junit-pioneer.org/docs/system-properties/#thread-safety

Reply:
So, should we preserve tests with `@SetSystemProperty` for now? 🤔

Reply:
I was referring to the other cases using `System.setProperty` 😅    I think using either of the junit-pioneer annotations should be fine 👍